lua/midi.lua

```diff
@@ -51,6 +61,64 @@ function Midi:send(data)
   midi_send(self.dev, data)
 end
 
+--- send midi event to named device
+function Midi.send_named(name, data)
+  if Midi.reverse[name] then
+    Midi.devices[Midi.reverse[name]]:send(data)
+  end
+end
+
+--- send midi event to all devices
+function Midi.send_all(data)
+  for id,device in pairs(Midi.devices) do
+    midi_send(device.dev, data)
+  end
+end
+
+
+--- subscribe
+function Midi.subscribe(callback, name)
+  if not name then
+    table.insert(Midi.broadcast, callback)
+    return function(data) midi.send_all(data) end
+  else
+    if not Midi.callbacks[name] then
+      Midi.callbacks[name] = {}
+      Midi.reverse[name] = nil -- will be overwritten with device insertion
+    end
+    table.insert(Midi.callbacks[name], callback)
+    return function(data) midi.send_named(name, data) end
+  end
+end
+
+--- unsubscribe
+function Midi.unsubscribe(callback)
+  for i,v in pairs(Midi.broadcast) do
```

Comment:
again on `pairs` ... will an arbitrary unsubscription order ever be surprising?  (in general i'd expect unsubscription to happen in a deterministic order the reverse of subscription but maybe in practice this doesn't matter?)